ADDING_NEW_CLIENTS.md

```diff
@@ -0,0 +1,108 @@
+## How to Add a New Client Library
+
+### Overview
+
+This document outlines how to add a new client library to `gcloud-java`.  New client libraries should be submodules located in a folder within the main repository, built using Maven.  A client library should contain the following items:
+
+* An API layer, with which users will interact.  This includes model objects and a service class. 
+* An SPI layer, which translates gcloud-java API calls into RPCs using an autogenerated client library.  In almost all use cases, the user will not directly interact with this code.  Separating this code from the API layer allows the API layer to remain stable despite changes to the autogenerated libraries used.
+* A test helper class, which allows users to easily interact with a local emulator (if possible).  If there is no emulator available and the service is too complex to create a mock, then this helper should facilitate separation of test data from other user data and enable easy cleanup.
+* Tests, including unit tests and integration tests.
+* A command line example application.
+* Documentation, which is comprised of READMEs, Javadoc, and code snippets.
+
+### Components of a new client library
+
+#### API layer
+
+Before starting work on the API layer, write a design document and provide sample API code, either in the design document or as a pull request tagged with the "don't merge" label.  As part of the design process, be sure to examine the Google Cloud service API and any implementations provided in other gcloud-* language libraries.  Solicit feedback from other contributors to the repository.  Client libraries should be low-level while minimizing boilerplate code needed by API users.  The library should also be flexible enough to be used by higher-level libraries.  For example, Objectify should be able to use `gcloud-java-datastore`.
```

Comment:
We should mention that the design document is essentially showing the intended interaction of the user with the new gcloud-java module. It may not be obvious for a person who never saw an example.